src/compiler.ts

```diff
@@ -1084,8 +1084,16 @@ export class Compiler extends DiagnosticEmitter {
               (<EnumValue>member).identifierNode.range.atEnd
             );
           }
+          var previousValueExpr: ExpressionRef;
+          if (isInline) {
+            let value = previousValue.constantIntegerValue;
+            assert(!i64_high(value));
+            previousValueExpr = module.i32(i64_low(value));
```

Comment:
Now that I look at it again, we can even precompute the add here in the inline case using `i64_add(..., i64_new(1))` and set `initExpr` directly.